### PR TITLE
fix: Introduce default headers

### PR DESCRIPTION
**Pull request by marikaner**
## Context

OData requests did only allow setting the content type header as a default header. We hardcoded other headers (currently 'accept') in the header function. This PR allows to set all default headers directly on the config and removes the later addition of header.

## Definition of Done

Please consider all items and remove only if not applicable.

- [x] Tests created/adjusted for your changes.
- [x] Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [x] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
- [x] If applicable: Properly documented (JSDoc of public API)
- [ ] If applicable: Check if `node run doc` still works.

### CODE REVIEW

File: packages/core/src/header-builder/headers-util.ts [hunk 1]
```diff
@@ -73,3 +73,19 @@ export function replaceDuplicateKeys(
...
     .reduce((replaced, header) => ({ ...replaced, ...header }), {});
 }
+
+/**
+ * Create a header object by merging two header objects, where the custom headers take precedence. .
+ * @param headers - A base header object that contains the headers that will be compared with `customHeaders`.
+ * @param customHeaders - A header object to be compared with headers. Only headers present in `headers` will be compared.
+ * @returns - An object containing all keys from both the header objects, where headers present in the `customHeaders` are replaced. Note that the case (upper / lower) used by `customHeaders` will be used.
+ */
+export function mergeHeaders(
```
Comment (FrankEssenberger) [inline packages/core/src/header-builder/headers-util.ts:83]: Could this methed be used in the [the odata-request](https://github.com/SAP/cloud-sdk/blob/defe3b7e48abab45b225c9e829d2dc3d1c59599b/packages/core/src/odata/common/request/odata-request.ts#L197) and also other places where I see the replaceDuplicateKeys?

File: packages/core/src/odata/common/request/odata-request-config.ts [hunk 2]
```diff
@@ -14,24 +15,44 @@ export abstract class ODataRequestConfig {
...
    * @param method - HTTP method of the request
    * @param defaultServicePath - default path of the according service
-   * @param contentType - The content type of the request
+   * @param defaultHeadersOrContentType - The default headers of the given request as an object. When passing a string only set the content type header will be set. Setting the content type only is deprecated since v1.30.0.
    */
   constructor(
     public method: RequestMethodType,
     readonly defaultServicePath: string,
-    readonly contentType = 'application/json'
+    defaultHeadersOrContentType?: Record<string, any> | string
```
Comment (FrankEssenberger) [inline packages/core/src/odata/common/request/odata-request-config.ts:43]: Could you overload the constructor and deprecate the one with the string argument. Then it is easier to find and remove later on.

File: packages/core/src/odata/common/request/odata-request.ts [hunk 3]
```diff
@@ -194,19 +194,18 @@ export class ODataRequest<RequestConfigT extends ODataRequestConfig> {
    * @returns Key-value pairs where the key is the name of a header property and the value is the respective value
    */
   defaultHeaders(): Record<string, any> {
-    const defaultHeaders = replaceDuplicateKeys(
-      filterNullishValues({
-        accept: 'application/json',
-        'content-type': this.config.contentType
+    const customDefaultHeaders = Object.keys(this.config.defaultHeaders).reduce(
```
Comment (FrankEssenberger) [inline packages/core/src/odata/common/request/odata-request.ts:197]: It took me a moment to realize what happens here. What about adding a new util method `getHeaders` where you can put in a list of keys and get the headers related ignoring case of the key in the search.

File: packages/core/src/header-builder/headers-util.ts (hunk 1 above)
Comment (FrankEssenberger) [reply packages/core/src/header-builder/headers-util.ts:83]: Forget this comment. I was not checking the end.

File: packages/core/src/odata/common/request/odata-request-config.ts (hunk 2 above)
Comment (marikaner) [reply packages/core/src/odata/common/request/odata-request-config.ts:43]: I did this initially but it is against our lint rules. I can turn them off, but I am not sure if it is possible to define multiple docs for this. I'll check.

File: packages/core/src/odata/common/request/odata-request.ts (hunk 3 above)
Comment (marikaner) [reply packages/core/src/odata/common/request/odata-request.ts:197]: I can do that.

**Review by FrankEssenberger — APPROVED** [review]
LGTM as discussed we consider for version 2.0 to create a rambda like util with getKeyValueIgnoreingCase or something like this which is then used by the header.

File: packages/core/src/odata/common/request/odata-request-config.ts (hunk 2 above)
Comment (marikaner) [reply packages/core/src/odata/common/request/odata-request-config.ts:43]: Forgot to add it, but will do on the next PR.

**Pull request merged**